compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/desugar/ASTBuilderUtil.java

```diff
@@ -724,19 +723,19 @@ public static BLangMatchExpression createMatchExpression(BLangExpression expr) {
         return matchExpr;
     }
 
-    public static BLangFieldBasedAccess createFieldAccessExpr(BLangVariableReference varRef, BLangIdentifier field) {
+    public static BLangFieldBasedAccess createFieldAccessExpr(BLangExpression varRef, BLangIdentifier field) {
```

Comment:
Can we accept `BLangAccessibleExpression`s here?

Reply:
Fixed in 004d3a3